radix-engine/src/system/system.rs

```diff
@@ -2869,99 +2884,7 @@ where
     feature = "std",
     catch_unwind(crate::utils::catch_unwind_system_panic_transformer)
 )]
-impl<'a, Y, V> ClientCryptoUtilsApi<RuntimeError> for SystemService<'a, Y, V>
-where
-    Y: KernelApi<System<V>>,
-    V: SystemCallbackObject,
-{
-    #[trace_resources(log=message.len())]
-    fn bls12381_v1_verify(
-        &mut self,
-        message: &[u8],
-        public_key: &Bls12381G1PublicKey,
-        signature: &Bls12381G2Signature,
-    ) -> Result<u32, RuntimeError> {
-        self.api.kernel_get_system().modules.apply_execution_cost(
-            ExecutionCostingEntry::Bls12381V1Verify {
-                size: message.len(),
-            },
-        )?;
-        Ok(verify_bls12381_v1(message, public_key, signature) as u32)
-    }
-
-    // Trace average message length and number of public_keys
-    #[trace_resources(log={pub_keys_and_msgs.iter().flat_map(|(_, msg)| msg).count()/pub_keys_and_msgs.len()},log=pub_keys_and_msgs.len())]
```

Comment:
`trace_resource` macro should be moved to `aggregate_verify_bls12381_v1()` function.

Reply:
good point!